src/rng.rs

```diff
@@ -131,6 +144,53 @@ impl<'d> Rng<'d> {
         self.info.regs.mctl().write(|w| w.rst_def().set_bit().prgm().set_bit());
     }
 
+    fn fill_chunk_inner(&mut self, chunk: &mut [u8]) -> Result<(), Error> {
+        let mut entropy = [0; 16];
+
+        for (i, item) in entropy.iter_mut().enumerate() {
+            *item = self.info.regs.ent(i).read().bits();
+        }
+
+        if entropy.contains(&0) {
+            return Err(Error::SeedError);
+        }
+
+        sw_entropy_test(&entropy)?;
+
+        // SAFETY: entropy is the same for input and output types in
+        // native endianness.
+        let entropy: [u8; 64] = unsafe { core::mem::transmute(entropy) };
```

Comment:
Since this relies on `entropy` being `[u32; 16]`, I think it would be helpful to mark that explicitly on the type instead of relying on the fact that integer literals default to `i32`